core/src/parser/statement.rs

```diff
@@ -107,4 +107,55 @@ impl<'src> Parser<'src> {
 
         Ok(id)
     }
+
+    #[inline(always)]
+    pub fn break_statement(&mut self) -> Result<Node<'src>> {
+        let tok = peek!(self);
+        let statement = Item::BreakStatement {
+            label: match tok {
+                Semicolon | EndOfProgram => None,
+                _ => {
+                    if self.lexer.asi() {
+                        None
+                    } else {
+                        let label = expect_identifier!(self);
+                        let id = self.store(Item::Identifier(label.into()).at(0, 0));
+                        Some(id)
+                    }
+                }
+            }
+        };
+
```

Comment:
Missing `expect_semicolon!(self)` here from the old parser implementation. The parser peeks at next token and will already check for `Semicolon`, and even do an ASI check, but it never consumes it.

Alternatively, and preferably, this could also be improved compared to the old parser by separating the `Semicolon | EndOfProgram` pattern into separate ones:

```rust
        let statement = Item::BreakStatement {
            label: match tok {
                EndOfProgram => None,
                Semicolon => {
                    self.consume();
                    None
                },
                _ => {
                    if self.lexer.asi() {
                        None
                    } else {
                        let label = expect_identifier!(self);
                        let id = self.store(Item::Identifier(label.into()).at(0, 0));
                        Some(id)
                    }
                }
            }
        };
```